src/main/java/command/Command.java

```diff
@@ -0,0 +1,184 @@
+                i++; // skip "/by" or "(by:"
+                while (i < strArr.length) {
+                    sb.append(strArr[i]);
+                    sb.append(" ");
+                    i++;
+                }
+                String by = sb.toString().substring(0, sb.length() - 1);
+
+                Task deadline = new Deadline(description, by);
+                list.add(deadline);
+                Ui.printAddedTask(deadline, list.getUncompletedSize());
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new SundayException("OOPS!!! The description of a deadline cannot be empty.");
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new SundayException("OOPS!!! The description of this deadline is invalid.");
+            } catch (ParseException e) {
+                throw new SundayException(
+                        "OOPS! It appears the deadline given was not of the format dd/mm/yyyy hhmm");
+            }
+        }
+    },
+    EVENT ("event") {
+        @Override
+        public void execute(String input) throws SundayException {
+            try {
+                input = input.substring(1);
+                String[] strArr = input.split(" ");
+                int j = 0;
+                StringBuilder sb = new StringBuilder();
+                while (!(strArr[j].equals("/from") || strArr[j].equals("(from:"))) {
+                    sb.append(strArr[j]);
+                    sb.append(" ");
+                    j++;
+                }
+                String description = sb.toString().substring(0, sb.length() - 1);
+
+                sb.setLength(0);
+                j++; // skip "/from" or "(from:"
+                while (!(strArr[j].equals("/to") || strArr[j].equals("to:"))) {
+                    sb.append(strArr[j]);
+                    sb.append(" ");
+                    j++;
+                }
+                String start = sb.toString().substring(0, sb.length() - 1);
+
+                sb.setLength(0);
+                j++; // skip "/to" or "to:"
+                while (j < strArr.length) {
+                    sb.append(strArr[j]);
+                    sb.append(" ");
+                    j++;
+                }
+                String end = sb.toString().substring(0, sb.length() - 1);
+                Task event = new Event(description, start, end);
+                list.add(event);
+                Ui.printAddedTask(event, list.getUncompletedSize());
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new SundayException("OOPS!!! The description of an event cannot be empty.");
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new SundayException("OOPS!!! The description of this event is invalid.");
+            } catch (ParseException e) {
+                throw new SundayException(
+                        "OOPS! It appears the deadline given was not of the format dd/mm/yyyy hhmm");
+            }
```

Comment:
As I notice there are a few lines of code that are similar to each other (have similar functionality), do you think creating another helper method to deal with duplicate lines will help reducing the number of code lines and improve  understandability?

Reply:
As this implementation is specific to the Event task, I didn't see a need to abstract away the similar lines of code, however I do agree that the duplicity of my code could affect understandability!